test/session-pool.js

```diff
@@ -16,2022 +16,1575 @@
+            assert.strictEqual(sessionPool.options.writes, 0.5);
+            assert.strictEqual(sessionPool.options.min, 0);
+            assert.strictEqual(sessionPool.options.maxReads, 40);
+            assert.strictEqual(sessionPool.options.maxWrites, 60);
+            assert.strictEqual(sessionPool.options.minReads, 0);
+            assert.strictEqual(sessionPool.options.minWrites, 0);
+          });
+        });
 
-    it('should create an acquire queue with 1 concurrency', function() {
-      pQueueOverride = function(options) {
-        return options;
-      };
+        it('should override DEFAULT for minReads and minWrites', function() {
+          return sessionPool.close().then(function() {
+            sessionPool = new SessionPool(DATABASE, {
+              writes: 0.5,
+              maxReads: 40,
+              minReads: 41,
+              minWrites: 61,
+              min: 70,
+            });
+            assert.strictEqual(sessionPool.options.acquireTimeout, Infinity);
+            assert.strictEqual(sessionPool.options.concurrency, 10);
+            assert.strictEqual(sessionPool.options.maxWait, 50);
+            assert.strictEqual(sessionPool.options.idlesAfter, 50);
+            assert.strictEqual(sessionPool.options.max, 100);
+            assert.strictEqual(sessionPool.options.writes, 0.5);
+            assert.strictEqual(sessionPool.options.min, 70);
+            assert.strictEqual(sessionPool.options.maxReads, 40);
+            assert.strictEqual(sessionPool.options.maxWrites, 60);
+            assert.strictEqual(sessionPool.options.minReads, 40);
+            assert.strictEqual(sessionPool.options.minWrites, 60);
+          });
+        });
+
+        it('should override DEFAULT for minReads and minWrites', function() {
+          return sessionPool.close().then(function() {
+            sessionPool = new SessionPool(DATABASE, {
+              writes: 0.5,
+              maxReads: 40,
+              min: 70,
+            });
+            assert.strictEqual(sessionPool.options.acquireTimeout, Infinity);
+            assert.strictEqual(sessionPool.options.concurrency, 10);
+            assert.strictEqual(sessionPool.options.maxWait, 50);
+            assert.strictEqual(sessionPool.options.idlesAfter, 50);
+            assert.strictEqual(sessionPool.options.max, 100);
+            assert.strictEqual(sessionPool.options.writes, 0.5);
+            assert.strictEqual(sessionPool.options.min, 70);
+            assert.strictEqual(sessionPool.options.maxReads, 40);
+            assert.strictEqual(sessionPool.options.maxWrites, 60);
+            assert.strictEqual(sessionPool.options.minReads, 40);
+            assert.strictEqual(sessionPool.options.minWrites, 60);
+          });
+        });
 
-      sessionPool = new SessionPool(DATABASE);
-      assert.deepEqual(sessionPool.acquireQueue_, {
-        concurrency: 1,
+        it('should override DEFAULT for minReads and minWrites', function() {
```

Comment:
Looks like we have duplicates of this test (and perhaps a couple others). Is there a reason for it? If we're testing for something specific here can we alter the test description to reflect that?

Reply:
Fixed it. There are not duplicates. I updated the description